scarlet/renderer.py

```diff
@@ -237,6 +240,69 @@ def convolve(self, model, convolution_type=None):
         return result
 
 
+class ShiftConvolutionRenderer(ConvolutionRenderer):
+
+    def __init__(self, data_frame, model_frame, convolution_type="fft", padding=10, shift = np.zeros(2)):
+
+        if shift is not None:
+            shift = Parameter(
+                shift, name="psf_shift", step=1.e-2)
+        super().__init__(data_frame,
+                         model_frame,
+                         shift,
+                         convolution_type=convolution_type,
+                         padding=padding)
+
+    def convolve(self, model, convolution_type=None, shift = None):
+        """Convolve the model in a single band
+        """
+        if convolution_type is None:
+            convolution_type = self._convolution_type
+        if shift is not None:
+            kernel = fft.shift(self.diff_kernel.image,
+                           shift,
+                           fft_shape=None,
+                           axes=(-2, -1),
+                           return_Fourier=True)
+        else:
+            kernel = self.diff_kernel.image
+        if convolution_type == "real":
+            result = convolve(model, kernel, self.convolution_bounds)
+        elif convolution_type == "fft":
+            result = fft.convolve(
+                fft.Fourier(model), kernel, axes=(1, 2)
+            ).image
+        else:
+            raise ValueError(
+                "`convolution` must be either 'real' or 'fft', got {}".format(
+                    convolution_type
+                )
+            )
+
+        return result
+
+    def __call__(self, model, *parameters):
+        self.transform = self.get_model(*parameters)
+        return self.transform(model, *parameters)
+
+
+    def get_model(self, *parameters):
+
+        def transform(model, *parameters):
+            # restrict to observed channels
+            model_ = self.map_channels(model)
+            # convolve observed channels
+            shift = self.get_parameter(0, *parameters)
+            print('zizizi',shift)
+            model_ = self.convolve(model_, shift=shift)
+            # adjust spatial shapes
+            model_ = match_shape(model_, self.data_frame, self.slices)
+            return model_
+
+
+        return transform
+
+
```

Comment:
Seeing this implementation, I think I should change my stance on this being a separate class. It's much simpler to add a single `shift=None` parameter to `ConvolutionRenderer` instead of dubplication almost all of it's contents, for the one thing that really happens (namely line 262)

Reply:
Done, Let's wait to see if tests pass and we can merge. Warning though, you're going to hate how I unfied the classes. I removed the original one and made the `ShiftConvolutionRenderer` the new `ConvolutionRenderer`. It's git-dirty.

Reply:
Tests fail, wait till I fix it.